tests/unit-tests/region-owner.cxx

```diff
@@ -0,0 +1,42 @@
+#include "doctest/doctest.h"
+
+#include <ipr/traversal>
+#include <ipr/impl>
+#include <ipr/utility>
+
+const ipr::Type* get_enclosing_udt_if_any(const ipr::Region& current_region)
+{
+    auto* region = &current_region;
+    for (;;)
+    {
+        switch (region->owner().category)
+        {
+        case ipr::Category_code::Class:
+        case ipr::Category_code::Enum:
+        case ipr::Category_code::Union:
+            return ipr::util::view<ipr::Type>(region->owner());
+
+        case ipr::Category_code::Block:
+        case ipr::Category_code::Namespace:
+            return nullptr;
+
+        default:
+            region = &region->enclosing();
+        }
+    }
+}
+
+TEST_CASE("Region-owner user") {
+  using namespace ipr;
+  impl::Lexicon lexicon{};
+  impl::Module module{lexicon};
+  impl::Interface_unit unit{lexicon, module};
+
+  // the helper above does not consider namespace to be udt type
+  auto* not_found = get_enclosing_udt_if_any(*unit.global_region());

```

Comment:
Why?

Reply:
For example, the helper is trying to find the nearest class or union scope. From C++ perspective, namespace is not a user defined type.

If you find this example objectionable, I can write up the one that looks for nearby block or namespace scope:

[dcl.type.elab]/3 ... The target scope of E is the nearest enclosing
namespace or block scope.

Reply:
> For example, the helper is trying to find the nearest class or union scope. From C++ perspective, namespace is not a user defined type.

From ISO C++ perspective, a union is also a class type.

A UDT is a user-defined type, and that includes enums.  From IPR perspective, a namespace is also a UDT.  In the description above, it sounds like you meant `get_enclosing_class_type_if_any`?

> If you find this example objectionable, I can write up the one that looks for nearby block or namespace scope

I think we should keep the terminology consistent within the IPR and examples, as people tend to easily extrapolate from what they see.